### ISSUE TITLE
Change text color when doing getImage()

### ISSUE THREAD

**Issue by milosevicn** [amcharts/amcharts4#4093]
I am using `am4themes_dark` for my charts, which results in having black background when when doing `getImage()`. To remove background color, there is an option under `exporting` for that (`chart.exporting.backgroundColor = am4core.color("#f00", 0);`). But that leaves me with white text on transparent background, and I am using those images to print them on PDF, and therefore white text becomes invisible there.

Is there a universal way to change text color, no matter which chart type I am doing it for, through `exporting` methods/properties?

**Comment by martynasma** [comment]
Maybe this will help:
https://www.amcharts.com/docs/v4/tutorials/modifying-chart-for-export/

**Comment by milosevicn** [comment]
Thanks, I was considering that also, but the problem is `getImage` function does not trigger export events.

Could I somehow manually trigger `exportstarted` and later `exportfinished` events?

**Comment by martynasma** [comment]
You don't need those events since you are triggering export yourself. You can make the changes before calling `getImage()`.

Please note that you might still need to add elements that you change to `validateSprites`:
https://www.amcharts.com/docs/v4/tutorials/modifying-chart-for-export/#Validating_updates

**Comment by milosevicn** [comment]
@martynasma Thank you, I see now, your answer helped me to understand why I didn't need those events.

And here is the solution which worked for me:

**TL;DR** - you have to recreate the charts with dark theme unused before calling `getImage()`.

---------------------------------------------------------------------------------------------------------
I had to split `exportToPdf` into two functions. 
First one is being triggered on click and it turns off dark theme and emits to parent to inform it that charts need to be recreated:

```
prepareForExport()
{
   if (!this.charts || this.charts.length === 0) return;
                
   am4core.unuseTheme(am4themes_dark);
   this.$emit('reload-chart', () => this.exportToPdf());
}
```
The important part here is that I am passing callback function as a parameter, which is calling a second function, where you can put `getImage()` logic:
```
exportToPdf: function()
{
   // this.charts is array of amCharts; the array may contain different types of charts
   if (!this.charts || this.charts.length === 0) return;
      
   let promises = [];
   for (let i in this.charts) {
        // this.charts[i].exporting.backgroundColor = am4core.color("#f00", 0); // Transparent background
        promises.push(this.charts[i].exporting.getImage('png'));
   }

    Promise.all( promises ).then((chart_images) => {
        am4core.useTheme(am4themes_dark);
        this.$emit('reload-chart');
        this.startPdfExport(chart_images);
    });
},
```

So, what I am doing inside parent, with async/await logic I am waiting till charts are recreated without dark theme, and only then I am calling the function which I passed as a paremeter.
After that, once `Promise.all` part is finished and goes to `.then()`, I am once again emitting to parent to inform it that charts should be recreated again to retrieve the dark theme look.

I hope someone will find this useful! 

P.S.
I know that passing functions as parameter through emit is anti-pattern in Vue, but that's the solution I found on the web and couldn't remember a better one.
